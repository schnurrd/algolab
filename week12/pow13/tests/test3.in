30
8 4
-1 1015 2890 1305 2640
46857 0 0 0 0
0 0 0 0 0
-1 22986 23427 22887 23775
382 0 0 0 0
-1 82 162 227 144
-1 50 37 35 28
-1 133 127 176 204
1 0
5 4
2 0
4 0
3 1
7 1
6 4
9031 0 0
8391 0 0
5441 0 0
6401 0 0
6 3
-1 2394 1100 2304
-1 15382 15174 15307
-1 635 533 605
36729 0 0 0
-1 9253 9251 9091
886 0 0 0
4 3
1 3
3 0
2 5
5 0
7577 0 0
6084 0 0
6240 0 0
1 5
-1 12356 12618 12673 12602 12304
0 0 0
0 0 0
0 0 0
0 0 0
0 0 0
12 2
-1 1315 970
-1 21451 22725
0 0 0
153 0 0
80 0 0
-1 238 174
-1 47 114
-1 247 269
258 0 0
22345 0 0
-1 158 149
-1 66 37
2 0
10 3
3 0
4 0
5 9
6 4
8 0
9 0
7 8
11 9
1 9
4531 0 0
7116 0 0
9 2
-1 2513 2231
-1 20 66
22929 0 0
311 0 0
43 0 0
-1 325 235
0 0 0
-1 30 33
-1 23018 22840
1 4
2 0
3 0
4 0
5 3
6 0
8 2
7 3
5801 0 0
5974 0 0
21 2
-1 2593 400
3216 0 0
-1 7290 8121
-1 84 95
0 0 0
2336 0 0
0 0 0
1001 0 0
22224 0 0
-1 1395 1450
-1 228 430
45536 0 0
-1 11 69
-1 8808 8819
89 0 0
-1 115 192
-1 36771 38891
40 0 0
11713 0 0
-1 1065 937
-1 2491 2182
19 7
13 8
1 0
16 11
3 14
4 0
5 13
6 0
7 13
8 0
9 1
2 11
11 10
12 17
14 9
10 18
20 5
15 1
17 9
18 13
3583 0 0
3480 0 0
18 4
-1 3758 3089 3357 1734
0 0 0 0 0
87 0 0 0 0
0 0 0 0 0
1185 0 0 0 0
-1 144 186 142 226
13699 0 0 0 0
256 0 0 0 0
-1 3589 3817 3796 3789
-1 636 619 530 585
-1 175 137 142 59
-1 6674 7447 6559 6718
-1 483 496 457 456
55619 0 0 0 0
7495 0 0 0 0
-1 13904 13619 13550 13954
4370 0 0 0 0
349 0 0 0 0
1 11
2 11
11 6
3 15
4 15
5 17
6 12
7 0
15 13
9 4
8 14
12 16
13 0
14 0
10 7
16 0
17 2
3928 0 0
3590 0 0
2045 0 0
1688 0 0
23 4
-1 1433 1828 2922 1759
639157 0 0 0 0
0 0 0 0 0
-1 98 184 160 33
-1 132 204 202 163
31 0 0 0 0
-1 2300 1694 1289 2197
-1 7 9 6 41
732558 0 0 0 0
-1 365 329 300 550
0 0 0 0 0
-1 225 115 696 22
-1 309516 324648 327892 314714
-1 8681 9150 8874 7651
-1 28 0 41 36
3740 0 0 0 0
743024 0 0 0 0
-1 305 262 356 252
-1 1184 938 1222 1083
529 0 0 0 0
733960 0 0 0 0
-1 2073 1835 2481 2171
729258 0 0 0 0
1 21
2 0
3 8
4 20
5 0
6 15
7 5
8 4
9 1
10 21
11 19
14 16
15 21
13 22
16 0
18 16
19 13
20 18
12 1
17 8
21 22
22 17
71995 0 0
80834 0 0
70619 0 0
2807 0 0
17 5
-1 3473 3085 3545 1751 2657
0 0 0 0 0 0
196264 0 0 0 0 0
167792 0 0 0 0 0
0 0 0 0 0 0
284 0 0 0 0 0
-1 69 17 39 50 44
0 0 0 0 0 0
1669 0 0 0 0 0
109 0 0 0 0 0
-1 655 618 723 634 709
-1 2286 2888 2258 2294 2683
-1 161 179 49 38 142
-1 77942 77416 78145 80361 78665
0 0 0 0 0 0
3535 0 0 0 0 0
-1 1171 1192 1464 1695 1548
1 10
2 11
3 10
4 11
13 2
5 11
6 9
7 16
8 0
11 3
9 0
10 8
12 5
14 0
15 11
16 15
5530 0 0
5437 0 0
2353 0 0
6666 0 0
2887 0 0
21 2
-1 1920 2621
-1 5386 5741
469 0 0
13470 0 0
0 0 0
31821 0 0
-1 308 346
-1 750 783
327 0 0
0 0 0
-1 138 99
9543 0 0
24855 0 0
8721 0 0
-1 32738 30905
681 0 0
-1 9027 9837
-1 262 440
-1 679 683
-1 3587 3494
-1 90 132
1 3
17 2
2 19
3 0
4 19
5 7
16 11
6 8
7 13
8 1
9 1
10 2
11 19
12 0
13 1
14 5
15 19
19 12
18 15
20 11
2663 0 0
4591 0 0
22 4
-1 1383 1870 297 705
-1 14 40 16 2
159153 0 0 0 0
0 0 0 0 0
400 0 0 0 0
234 0 0 0 0
-1 147 139 124 45
331 0 0 0 0
-1 7214 7223 6389 6614
-1 83 162 95 128
-1 19000 18996 19445 19419
8535 0 0 0 0
-1 98 37 17 5
-1 272 442 383 265
15853 0 0 0 0
908 0 0 0 0
36 0 0 0 0
-1 126 90 96 60
-1 3980 4454 4422 4215
-1 4 12 35 40
-1 157 178 260 205
78 0 0 0 0
1 16
8 14
2 0
3 10
4 17
5 10
6 15
7 10
9 5
11 8
12 21
18 11
10 2
14 10
15 2
16 0
17 7
13 15
19 7
20 4
21 10
1437 0 0
2611 0 0
1521 0 0
4964 0 0
21 4
-1 3461 3164 3843 3755
431 0 0 0 0
0 0 0 0 0
-1 6 49 28 5
0 0 0 0 0
-1 134 131 147 72
222 0 0 0 0
36839 0 0 0 0
-1 256 202 161 244
-1 301 464 389 321
-1 79 28 106 62
-1 18567 18523 17871 18717
0 0 0 0 0
-1 38 15 32 33
-1 57 90 37 108
-1 56 86 2 40
59 0 0 0 0
103 0 0 0 0
-1 49 39 41 41
979 0 0 0 0
238 0 0 0 0
1 0
2 0
11 7
10 6
3 17
4 8
5 19
6 0
7 0
9 19
8 1
12 18
13 16
14 20
15 20
16 0
17 0
18 6
19 0
20 3
974 0 0
711 0 0
1717 0 0
1910 0 0
22 3
-1 1096 2491 235
292 0 0 0
0 0 0 0
411 0 0 0
-1 93 59 166
9365 0 0 0
9482 0 0 0
66388 0 0 0
-1 136 191 139
50397 0 0 0
-1 248 203 134
-1 77 47 48
-1 1516 1409 1636
-1 6207 5965 6241
-1 165 298 360
23 0 0 0
-1 13 4 16
-1 2494 2712 2735
86 0 0 0
-1 13 9 24
-1 33694 33663 32971
0 0 0 0
14 3
1 0
20 9
2 17
3 12
4 5
5 12
6 0
7 0
8 9
9 17
11 18
15 17
12 6
17 7
16 6
18 17
19 15
21 14
10 1
13 5
2389 0 0
3491 0 0
1194 0 0
20 5
-1 1274 1332 1236 1309 1148
4558 0 0 0 0 0
-1 7 9 36 44 10
2384 0 0 0 0 0
-1 1961 1781 1746 1733 1896
-1 76184 77430 80043 78410 77448
-1 1105 801 745 1135 982
0 0 0 0 0 0
100813 0 0 0 0 0
2714 0 0 0 0 0
731 0 0 0 0 0
-1 379 600 553 361 350
-1 93 51 88 135 79
-1 118 74 114 44 89
-1 1219 1033 1122 894 1161
194757 0 0 0 0 0
51160 0 0 0 0 0
309 0 0 0 0 0
-1 24 31 35 45 44
-1 69 69 26 123 21
1 0
2 10
3 11
4 1
6 3
7 0
8 12
9 2
12 16
10 0
11 8
5 15
19 16
13 17
14 9
15 8
16 0
17 0
18 17
4404 0 0
2864 0 0
4436 0 0
310 0 0
4087 0 0
20 4
-1 2341 2181 857 647
211 0 0 0 0
-1 258 69 226 266
41417 0 0 0 0
-1 75815 74526 72761 73894
317704 0 0 0 0
158852 0 0 0 0
1723 0 0 0 0
-1 20239 20722 21069 20058
442 0 0 0 0
0 0 0 0 0
-1 692 890 830 814
7148 0 0 0 0
-1 282 103 226 274
-1 146 214 153 165
-1 10 338 297 101
-1 1581 1498 1475 1606
0 0 0 0 0
409 0 0 0 0
-1 139 181 83 20
1 16
4 5
2 18
16 12
3 4
8 3
5 6
6 16
7 16
9 11
10 0
12 0
14 12
13 9
15 3
17 4
11 7
18 16
19 1
7973 0 0
13026 0 0
3749 0 0
11278 0 0
23 4
-1 2350 2506 1544 3724
-1 304 357 346 371
5364 0 0 0 0
-1 73 200 245 544
66 0 0 0 0
-1 1206 1466 1296 1574
-1 167 201 110 141
-1 2495 2810 2331 2302
1580 0 0 0 0
0 0 0 0 0
27275 0 0 0 0
-1 35 13 49 35
-1 51890 51765 54200 50212
842 0 0 0 0
-1 636 517 598 615
7551 0 0 0 0
1650 0 0 0 0
25474 0 0 0 0
-1 450 633 491 525
-1 69 97 72 69
-1 12377 13317 12249 13005
106804 0 0 0 0
-1 395 216 238 300
2 16
5 21
3 8
4 0
19 13
6 16
7 2
8 7
22 10
9 0
10 14
11 4
13 0
20 17
15 0
16 14
17 14
1 13
18 8
14 15
12 21
21 22
3412 0 0
1514 0 0
3096 0 0
3571 0 0
24 4
-1 3336 2711 1033 1192
145749 0 0 0 0
22167 0 0 0 0
-1 2940 2584 2768 2861
3628 0 0 0 0
590 0 0 0 0
-1 1884 1406 1630 961
514 0 0 0 0
529189 0 0 0 0
-1 243 224 274 288
390 0 0 0 0
-1 236 199 223 156
-1 431 95 323 527
-1 187 157 238 165
1112 0 0 0 0
-1 12 32 28 30
289870 0 0 0 0
84027 0 0 0 0
-1 580 698 523 424
-1 269797 255994 259672 272915
-1 183 219 241 137
-1 6305 5008 5880 6269
-1 135 85 136 77
-1 3583 3536 3016 3533
1 3
18 14
3 17
2 0
23 2
19 8
21 16
4 21
5 0
7 23
8 21
9 7
10 23
11 1
12 4
13 5
14 23
6 4
15 2
16 11
17 23
20 10
22 5
26214 0 0
24054 0 0
2818 0 0
20299 0 0
23 2
-1 1793 2276
47 0 0
4340 0 0
83 0 0
-1 924 476
-1 39195 39835
-1 21 74
-1 31 104
499 0 0
-1 219 181
-1 574 424
-1 73 56
0 0 0
95578 0 0
-1 215 187
-1 67 99
6728 0 0
-1 3695 4986
2449 0 0
264 0 0
-1 7007 6450
122174 0 0
-1 1180 1050
1 0
2 5
5 21
3 0
10 8
14 13
22 13
6 1
7 18
8 5
11 19
9 19
12 5
13 0
16 4
17 2
18 5
19 0
15 3
20 16
21 22
4 18
12588 0 0
10660 0 0
25 3
-1 404 2424 1616
-1 426 614 587
158994 0 0 0
-1 78 119 134
-1 0 6 28
-1 80038 83036 79739
341 0 0 0
163468 0 0 0
243306 0 0 0
-1 37 182 3
93 0 0 0
-1 632 687 819
-1 4514 5224 5129
1098 0 0 0
-1 442 331 455
-1 17 2 46
111 0 0 0
6168 0 0 0
213 0 0 0
-1 56 85 46
-1 119 274 34
0 0 0 0
-1 192 192 185
-1 298 288 96
-1 4198 3657 3253
1 13
11 7
2 11
3 8
4 7
6 12
7 0
24 17
8 12
9 16
10 0
13 12
14 17
12 2
15 7
16 5
17 12
18 5
19 10
20 18
5 8
23 6
21 12
22 13
9320 0 0
3239 0 0
14555 0 0
23 2
-1 2020 707
-1 78 542
171814 0 0
-1 128 110
-1 536 288
-1 192 47
-1 1167 639
5616 0 0
11277 0 0
-1 92 77
1387 0 0
-1 131 168
259 0 0
119 0 0
43856 0 0
3319 0 0
-1 3008 3631
-1 727 388
561 0 0
-1 1373 1488
-1 174021 169453
310 0 0
-1 251 207
1 21
17 10
19 7
20 2
2 6
16 15
3 13
4 18
5 12
7 0
8 19
9 8
10 19
11 18
22 8
12 19
13 19
14 22
15 17
18 5
6 14
21 20
13377 0 0
11661 0 0
25 3
-1 1696 2091 1310
865 0 0 0
-1 282 278 183
-1 86 100 15
231568 0 0 0
14779 0 0 0
4081 0 0 0
2201 0 0 0
-1 33 87 154
-1 206 92 293
-1 148838 156082 156177
-1 595 360 197
-1 2214 3054 2845
-1 3361 3469 3370
11531 0 0 0
-1 1737 1372 1293
576 0 0 0
58324 0 0 0
371 0 0 0
-1 1666 1092 1154
4056 0 0 0
-1 68 69 59
-1 550 366 526
-1 39 5 1
-1 741 715 766
1 13
2 18
3 5
4 9
5 13
19 6
10 4
6 10
7 13
8 17
9 17
12 20
15 7
13 14
22 1
14 0
16 22
17 21
18 13
11 16
20 19
21 5
23 14
24 6
7139 0 0
5931 0 0
1603 0 0
20 2
-1 8678 7438
-1 1023 1148
0 0 0
503 0 0
-1 38 117
-1 67 33
-1 414 732
1085 0 0
-1 8183 6815
-1 270 383
16832 0 0
-1 294 223
-1 440 351
50 0 0
-1 8083 8044
968 0 0
2012 0 0
271 0 0
326 0 0
7741 0 0
1 7
2 0
3 0
4 16
5 13
6 15
7 17
9 18
8 19
10 0
12 15
13 14
11 10
15 8
16 3
17 8
18 11
14 10
19 16
2473 0 0
1409 0 0
20 5
-1 2840 2074 2435 1254 2712
-1 81 9 26 52 19
33124 0 0 0 0 0
-1 214 177 251 213 225
-1 7476 7440 7613 7432 6984
0 0 0 0 0 0
2454 0 0 0 0 0
72779 0 0 0 0 0
2209 0 0 0 0 0
18472 0 0 0 0 0
540 0 0 0 0 0
-1 27883 28741 28652 29790 29389
0 0 0 0 0 0
-1 550 1026 1031 899 912
-1 917 1132 962 889 869
-1 2580 2567 2482 2515 2656
538 0 0 0 0 0
-1 198 182 266 270 160
-1 21 17 16 37 48
93 0 0 0 0 0
15 2
1 19
4 9
2 0
3 10
5 15
6 0
7 15
8 11
9 15
10 0
12 3
13 8
11 7
16 0
17 16
18 6
19 15
14 6
2532 0 0
3247 0 0
2722 0 0
837 0 0
3261 0 0
21 4
-1 1904 3794 2662 2296
-1 46 92 102 45
-1 88 44 36 79
-1 347 257 322 338
0 0 0 0 0
-1 7 14 19 33
-1 103 93 132 136
0 0 0 0 0
0 0 0 0 0
101 0 0 0 0
-1 11 1 13 27
123 0 0 0 0
41313 0 0 0 0
-1 20287 20813 20983 20544
-1 28 46 25 31
774 0 0 0 0
-1 115 112 101 115
-1 178 188 165 166
26 0 0 0 0
221 0 0 0 0
580 0 0 0 0
1 15
2 11
3 15
4 0
5 9
7 0
8 0
9 0
10 18
11 0
12 0
14 9
15 0
16 19
17 20
18 0
19 0
13 12
6 20
20 0
1011 0 0
2382 0 0
1966 0 0
780 0 0
20 2
-1 667 184
0 0 0
142 0 0
70798 0 0
-1 1520 1761
-1 10 24
293 0 0
75384 0 0
0 0 0
-1 374 205
1979 0 0
-1 84771 90617
87694 0 0
2007 0 0
-1 2077 1938
-1 2023 1936
-1 92 193
1172 0 0
-1 193 583
-1 604 651
11 12
1 14
2 4
3 19
4 3
5 7
6 4
7 0
8 4
9 17
14 13
10 18
12 4
13 19
16 2
17 6
15 10
18 17
19 7
5635 0 0
7938 0 0
21 3
-1 2810 1654 862
0 0 0 0
-1 2490 2308 2170
-1 180 271 126
-1 72037 69606 71083
288 0 0 0
62520 0 0 0
-1 838 993 762
-1 1149 1286 1163
-1 113 188 112
-1 2673 2907 3111
1296 0 0 0
-1 49 180 163
-1 28 12 14
-1 42 3 11
268 0 0 0
0 0 0 0
47175 0 0 0
3690 0 0 0
106363 0 0 0
1826 0 0 0
10 6
1 0
3 5
5 12
6 14
7 11
2 18
8 20
4 19
9 18
11 10
12 15
13 20
14 17
15 17
16 8
17 0
18 0
19 10
20 0
5732 0 0
7767 0 0
5532 0 0
24 4
-1 3730 3624 2663 2347
-1 21 10 45 46
-1 169 393 296 340
-1 15 73 63 40
-1 377 95 317 171
1027 0 0 0 0
10021 0 0 0 0
809 0 0 0 0
7511 0 0 0 0
50 0 0 0 0
4187 0 0 0 0
156 0 0 0 0
-1 1857 1855 1677 2025
41255 0 0 0 0
-1 444 421 377 376
-1 156 322 5 334
-1 40 22 37 1
156552 0 0 0 0
-1 1953 2037 2635 1750
-1 90 243 333 191
-1 315308 310703 317001 306141
-1 932 899 760 683
624576 0 0 0 0
-1 703 563 577 658
20 22
2 5
21 13
1 11
3 11
4 13
5 12
6 23
7 0
8 0
9 0
10 12
11 0
12 6
13 12
14 7
15 17
16 9
17 21
19 5
18 10
22 17
23 8
21568 0 0
10185 0 0
22441 0 0
19704 0 0
22 3
-1 5321 5595 4648
-1 92 64 37
104997 0 0 0
96 0 0 0
-1 476 596 339
588 0 0 0
-1 360 237 267
-1 680 1309 1306
30 0 0 0
-1 34517 34922 34970
-1 5 213 219
88 0 0 0
-1 445 471 469
-1 70 63 44
-1 544 509 516
218 0 0 0
2353 0 0 0
-1 13 6 41
-1 125 182 159
784 0 0 0
53883 0 0 0
665 0 0 0
1 3
2 12
3 0
4 16
5 9
6 21
8 0
9 2
10 15
11 0
13 11
15 9
12 20
16 5
17 8
14 19
18 21
19 0
20 0
21 9
7 16
3263 0 0
5676 0 0
6004 0 0
19 4
-1 12518 11765 12128 12547
-1 79 111 72 98
-1 119 155 135 123
0 0 0 0 0
2247 0 0 0 0
-1 951 937 1239 1007
0 0 0 0 0
266 0 0 0 0
-1 73 38 61 89
-1 35 29 31 11
54 0 0 0 0
0 0 0 0 0
8047 0 0 0 0
0 0 0 0 0
-1 41 35 12 86
53 0 0 0 0
-1 1298 1219 1344 1395
-1 17 33 34 25
217 0 0 0 0
1 4
2 7
3 0
4 0
5 4
6 16
7 0
8 18
9 15
10 0
11 16
12 0
13 0
14 18
15 0
17 10
16 12
18 16
1644 0 0
1167 0 0
912 0 0
1548 0 0
20 5
-1 303 1111 505 909 404
-1 1968 2023 1786 1789 1895
4269 0 0 0 0 0
-1 12920 13233 12322 13343 12758
284065 0 0 0 0 0
0 0 0 0 0 0
0 0 0 0 0 0
14289 0 0 0 0 0
-1 104 784 234 298 298
-1 413 313 504 662 247
-1 1018 939 743 592 865
-1 1309 1640 2117 1561 1911
-1 119998 125041 127385 129727 116275
-1 57 34 74 33 63
1928 0 0 0 0 0
130 0 0 0 0 0
309213 0 0 0 0 0
482 0 0 0 0 0
249601 0 0 0 0 0
-1 678 1238 1124 1224 998
2 10
3 4
12 16
4 1
5 1
1 7
6 1
7 0
8 14
9 14
10 18
11 2
13 15
14 17
15 1
16 19
17 19
18 3
19 18
17445 0 0
12988 0 0
14059 0 0
4248 0 0
12740 0 0
