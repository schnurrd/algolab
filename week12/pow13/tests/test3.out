0
0
0
0
0
0
0
Impossible!
Impossible!
0
0
0
0
0
Impossible!
0
Impossible!
0
Impossible!
Impossible!
Impossible!
0
0
0
0
0
Impossible!
0
0
Impossible!
